### PR TITLE
linter: fix issues since revision 75173a17cf

### PR DESCRIPTION
**Pull request by r4f4**

### CODE REVIEW

**Comment by r4f4** [comment]
Dropped https://github.com/openshift/installer/pull/6712/commits/83d0a6a2dd5dbfe735efa94f805727254413ac00 since it merged as part of #6601

**Review by barbacbd — COMMENTED** [review]
/lgtm

File: pkg/asset/agent/manifests/nmstateconfig_test.go [hunk 1]
```diff
@@ -180,12 +177,13 @@ func TestNMStateConfig_Generate(t *testing.T) {
...
-			if tc.expectedError != "" {
+			switch {
+			case tc.expectedError != "":
 				assert.ErrorContains(t, err, tc.expectedError)
-			} else if len(tc.expectedConfig) == 0 {
+			case len(tc.expectedConfig) == 0:
 				assert.NoError(t, err)
 				assert.Equal(t, tc.expectedConfig, asset.Config)
-			} else {
+			default:
```
Comment (patrickdillon) [inline pkg/asset/agent/manifests/nmstateconfig_test.go:186]: Perhaps overly prescriptive, but I kind of like it

File: pkg/asset/agent/manifests/util_test.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ var (
         state: up
         mac-address: 52:54:01:aa:aa:a1`
 
-	// config with route but no interface is invalid
+	// config with route but no interface is invalid.
```
Comment (patrickdillon) [inline pkg/asset/agent/manifests/util_test.go:59]: Hm... I didn't think godot would dot this kind of comment... I was curious what they meant by "top-level"

**Review by patrickdillon — COMMENTED** [review]
/lgtm
/approve

File: pkg/asset/agent/manifests/util_test.go (hunk 2 above)
Comment (r4f4) [reply pkg/asset/agent/manifests/util_test.go:59]: It seems the default is `declarations` [1]. But what that means is based on the Golang's AST [2]. I'll play around with the settings if we start getting too many of these.

[1] https://golangci-lint.run/usage/linters/#godot
[2] https://github.com/tetafro/godot/issues/22#issuecomment-896179124

File: pkg/asset/agent/image/agentimage.go [hunk 3]
```diff
@@ -84,7 +84,7 @@ func (a *AgentImage) PersistToFile(directory string) error {
 		return err
 	}
 
-	err = os.WriteFile(filepath.Join(directory, "rendezvousIP"), []byte(a.rendezvousIP), 0644)
+	err = os.WriteFile(filepath.Join(directory, "rendezvousIP"), []byte(a.rendezvousIP), 0o644) //nolint:gosec // no sensitive info
```
[the remark this answers is not in the document]
Comment (r4f4) [reply pkg/asset/agent/image/agentimage.go:87]: I'll follow up with the agent-installer team on this one. `gosec` always expects `0600` or less.

[the remark this answers is not in the document]
Comment (bfournie) [reply pkg/asset/agent/image/agentimage.go:87]: Follow up on slack discussion - I tested it using 0600 and its fine, I will create a PR to change this

**Comment by r4f4** [comment]
/override ci/prow/e2e-azure-ovn ci/prow/e2e-vsphere-ovn
Previous runs installed but timed out either during deprovision or during tests.

**Comment by r4f4** [comment]
/override ci/prow/e2e-azure-ovn ci/prow/e2e-gcp-ovn ci/prow/e2e-vsphere-ovn

**Comment by r4f4** [comment]
/override ci/prow/e2e-azure-ovn ci/prow/e2e-gcp-ovn ci/prow/e2e-vsphere-ovn ci/prow/e2e-aws-ovn

**Pull request merged**